src/mongoose_listener.erl

```diff
@@ -12,6 +12,8 @@
 -export([start/0, stop/0]).
 
 -callback start_listener(options()) -> ok.
+-callback listener_instrumentation(options()) -> [mongoose_instrument:spec()].
```

Comment:
I would call it `instrumentation`. One module shouldn't be a listener and an HTTP handler.